core/src/sql/number.rs

```diff
@@ -729,6 +729,34 @@ impl TryNeg for Number {
 	}
 }
 
+impl TryFloatDiv for Number {
+	type Output = Self;
+	fn try_float_div(self, other: Self) -> Result<Self, Error> {
+		Ok(match (self, other) {
+			(Number::Int(v), Number::Int(w)) => {
+				let quotient = (v as f64).div(w as f64);
+				if quotient.fract() != 0.0 {
+					return Ok(Number::Float(quotient));
+				}
+				Number::Int(
+					v.checked_div(w).ok_or_else(|| Error::TryDiv(v.to_string(), w.to_string()))?,
+				)
+			}
+			(Number::Float(v), Number::Float(w)) => Number::Float(v.div(w)),
+			(Number::Decimal(v), Number::Decimal(w)) => Number::Decimal(
+				v.checked_div(w).ok_or_else(|| Error::TryDiv(v.to_string(), w.to_string()))?,
+			),
+			(Number::Int(v), Number::Float(w)) => Number::Float((v as f64).div(w)),
+			(Number::Float(v), Number::Int(w)) => Number::Float(v.div(w as f64)),
+			(v, w) => Number::Decimal(
+				v.to_decimal()
+					.checked_div(w.to_decimal())
+					.ok_or_else(|| Error::TryDiv(v.to_string(), w.to_string()))?,
+			),
+		})
+	}
+}
+
```

Comment:
Could we have tests for this please in the bottom of this file under 
```
#[cfg(test)]
mod test {
 .... tests ...
}
```
You can initialise the numbers as you need and operate on them with the new functionality without setting up the database then.

Reply:
Thanks , i added a test now.